odl/solvers/util/callback.py

```diff
@@ -325,12 +328,18 @@ def __init__(self, *args, **kwargs):
         self.kwargs = kwargs
         self.fig = kwargs.pop('fig', None)
         self.display_step = kwargs.pop('display_step', 1)
+        self.saveto = kwargs.pop('saveto', None)
         self.iter = 0
 
     def __call__(self, x):
         """Show the current iterate."""
         if (self.iter % self.display_step) == 0:
-            self.fig = x.show(*self.args, fig=self.fig, **self.kwargs)
+            if self.saveto is None:
+                self.fig = x.show(*self.args, fig=self.fig, **self.kwargs)
+            else:
+                self.fig = x.show(*self.args, fig=self.fig,
+                                  saveto=(self.saveto + '_' + str(self.iter)),
```

Comment:
Suggestion: Offer the user an "advanced" variant where it is possible to specify a format string as `saveto` argument (something like `reco_iter_{:>03}` to enforce at least 3 digits), where we simply fill in the iteration number via `self.saveto.format(self.iter)`. That would require a way to determine if a format is present, which should be simple -- just check that there are 0 to 1 pairs `{` and `}`. For 0, replace the `saveto` by `saveto + '_{}' and then go ahead as in the "1" case.